docs/src/docs/asciidoc/user/performance.adoc

```diff
@@ -61,6 +61,8 @@ Size calculation is done using the https://github.com/ehcache/sizeof[SizeOf] lib
 This library uses multiple magic tricks to do so. It selects the fastest one for a given environment.
 Make sure of what is used to confirm you can't use a faster way on your platform.
 
+WARNING: Byte based heap resources are deprecated and will be removed in a future version.
```

Comment:
I would put this at the top of this section, immediately beneath the header, so that it's read first.

Reply:
Moved this to the beginning of paragraph.